model/user.go

```diff
@@ -693,7 +693,10 @@ func (u *User) IsSSOUser() bool {
 }
 
 func (u *User) IsOAuthUser() bool {
-	return u.AuthService == USER_AUTH_SERVICE_GITLAB
+	return u.AuthService == SERVICE_GITLAB ||
+		u.AuthService == SERVICE_GOOGLE ||
+		u.AuthService == SERVICE_OFFICE365 ||
+		u.AuthService == SERVICE_OPENID
```

Comment:
The only other place where this function is used is [here](https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/user_store.go#L191), and it doesn't appear to be Gitlab-specific.

Reply:
And where is `Update()` called from. Make sure you understand the impact of this change. It may even be fixing a bug. Luckily its only used in one place.

Reply:
The IsOAuthUser function was written 4-5 years ago and I'm guessing it just wasn't updated when we added more OAuth providers. As mentioned above, this function is only used in one other place, and it doesn't seem to be specific to Gitlab.
As for the `updateUser()` function, that is only used as the HTTP handler for the `users/{user_id}` endpoint. So the changes are specific to this endpoint.